packages/shared/lib/core/wallet/utils/convertToRawAmount.ts

```diff
@@ -19,13 +19,15 @@ export function convertToRawAmount(amount: string, tokenMetadata: TokenMetadata,
 function convertToRawAmountFromMetadata(amount: number, tokenMetadata: TokenMetadata, selectedUnit: string): Big {
     if (tokenMetadata?.standard === TokenStandard.BaseToken) {
         if (tokenMetadata.useMetricPrefix) {
-            return Big(amount * IOTA_UNIT_MAP?.[selectedUnit?.substring(0, 1)] ?? 0)
+            return Big(amount).mul(Big(10).pow(IOTA_UNIT_MAP?.[selectedUnit?.substring(0, 1)]?.decimalPlaces ?? 0))
```

Comment:
To help make this more readable, I'd create a `exponent` variable that call that with `.pow(exponent)`.